ProjectSingleton/ProjectSingleton/Program.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Threading;
+
+namespace ProjectSingleton
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // The client code.
+            Console.WriteLine(
+                "{0}\n{1}\n\n{2}\n",
+                "If you see the same value, then singleton was reused (yay!)",
+                "If you see different values, then 2 singletons were created (booo!!)",
+                "RESULT:"
+            );
+            Thread process1 = new Thread(() =>
+            {
+                TestSingleton("FOO");
+            });
+            Thread process2 = new Thread(() =>
+            {
+                TestSingleton("BAR");
+            });
+            process1.Start();
+            process2.Start();
+            process1.Join();
+            process2.Join();
+        }
+        public static void TestSingleton(string value)
+        {
+            Singleton singleton = Singleton.GetSingleton();
+            Console.WriteLine(singleton.Value);

```

Comment:
I see no value? your invoking the` singleton.Value` but its actually prints nothing!
the testing logic needs to be redesigned